### PR TITLE
Use Twirl instead of scalate

### PR DESCRIPTION
**Pull request by TorunR**
* Refactor Template accordingly
* Invokes the twirl template using reflection. This is ugly but it
  works.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -142,6 +142,7 @@ lazy val `OPAL` = (project in file("."))
   .settings((Defaults.coreDefaultSettings ++ Seq(publishArtifact := false)): _*)
   .enablePlugins(ScalaUnidocPlugin)
   .settings(
+    libraryDependencies += Dependencies.scalaCompiler(scalaVersion.value),
```
Comment (errt) [inline build.sbt]: Wouldn't it make sense to define this in the Dependencies file like the values for, e.g., bi and br?

File: project/SiteGeneration.scala [hunk 2]
```diff
@@ -1,6 +1,9 @@
 /* BSD 2-Clause License - see OPAL/LICENSE for details. */
-import sbt._
+
```
Comment (errt) [inline project/SiteGeneration.scala:4]: Formatting: Too many blank lines

File: project/SiteGeneration.scala [hunk 3]
```diff
@@ -11,12 +14,12 @@ import sbt.Keys.TaskStreams
 object SiteGeneration {
 
   def generateSite(
-      sourceDirectory: File,
-      resourceManaged: File,
-      streams: TaskStreams,
-      disassemblerJAR: File,
-      projectSerializerJAR: File
-  ): File = {
+                    sourceDirectory: File,
```
Comment (errt) [inline project/SiteGeneration.scala]: Formatting: Wrong indentation

File: src/site/defaultTemplate.scala.html [hunk 4]
```diff
@@ -1,11 +1,8 @@
-<%@ var title:String %>
-<%@ var content:String %>
-<%@ var toc:Traversable[(String,String)] %>
-<%@ var useBanner : Boolean = false %>
+@(title:String,content:String,toc:Traversable[(String,String)],useBanner:Boolean)
```
Comment (errt) [inline src/site/defaultTemplate.scala.html]: Formatting: Spaces after commas would make this more readable if that is allowed

**Review by errt — CHANGES_REQUESTED** [review]

**Comment by errt** [comment]
The code looks good to me now, but did you check the output? The HTML seems severely broken for the table of contents. E.g., now quotation marks are generated around anchor urls, headings are generated as
`<a href=/null.html>Text</a><br>` instead of `<br><b>Text</b><br>`
and there is lots of additional whitespace.
The commented out script has also reappeared in the output.

**Comment by TorunR** [comment]
I did check only an intermediate Version. I'll have to recheck

**Review by errt — APPROVED** [review]
LGTM now, thanks!

**Review by errt — CHANGES_REQUESTED** [review]
Nope, urls of TOC still without quotation marks...

**Review by errt — APPROVED** [review]

**Pull request merged**